opm/grid/cpgrid/processEclipseFormat.cpp

```diff
@@ -139,12 +145,32 @@ namespace cpgrid
                 thickness[i] = ecl_grid.getCellThicknes(i);
             }
             const double z_tolerance = ecl_grid.isPinchActive() ?  ecl_grid.getPinchThresholdThickness() : 0.0;
-            nnc = mp.process(thickness, z_tolerance, poreVolume, ecl_grid.getMinpvVector(), actnumData, opmfil, zcornData.data());
-            if (opmfil || nnc.size() > 0) {
+            nnc_cells_pinch = mp.process(thickness, z_tolerance, poreVolume, ecl_grid.getMinpvVector(), actnumData, opmfil, zcornData.data());
+            if (opmfil || nnc_cells_pinch.size() > 0) {
                 this->zcorn = zcornData;
             }
         }
 
+        NNCMaps nnc_cells;
+        // Add PINCH NNCs.
+        for (const auto& p : nnc_cells_pinch) {
+            auto low = std::min(p.first, p.second);
+            auto high = std::max(p.first, p.second);
+            nnc_cells[PinchNNC].insert({low, high});
+        }
+        // Add explicit NNCs.
+        for (const auto single_nnc : nncs.nncdata()) {
+            // Repeated NNCs will only exist in the map once
+            // (repeated insertions have no effect), and we make
+            // sure NNCs specified using either order of cells
+            // end up with the same {low, high} pair.
+            // The code that computes the transmissibilities is responsible
+            // for ensuring repeated NNC transmissibilities are added.
+            auto low = std::min(single_nnc.cell1, single_nnc.cell2);
```

Comment:
Thinking about why I would prefer to use
```
auto low  = single_nnc.cell1;
auto high = single_nnc.cell2;
if ( low > high ){
  std:swap(low, high)
}
```
makes me wonder how flow should behave if `low == high` holds. The parser just eats it. I guess that is an error in the deck and we should abort?

Reply:
I think explicit NNCs between a cell and itself should be flagged as an error in the parser. @joakim-hove do you agree?